samples/subsys/logging/logger/src/main.c

```diff
@@ -200,32 +207,61 @@ static void log_strdup_showcase(void)
  */
 static void performance_showcase(void)
 {
+/* Arbitrary limit when LOG_IMMEDIATE is enabled. */
+#define LOG_IMMEDIATE_TEST_MESSAGES_LIMIT 50
+
 	volatile u32_t current_timestamp;
 	volatile u32_t start_timestamp;
+	u32_t limit = COND_CODE_1(CONFIG_LOG_IMMEDIATE,
+			     (LOG_IMMEDIATE_TEST_MESSAGES_LIMIT),
+			     (CONFIG_LOG_BUFFER_SIZE / sizeof(struct log_msg)));
 	u32_t per_sec;
 	u32_t cnt = 0U;
 	u32_t window = 2U;
 
 	printk("Logging performance showcase.\n");
+	wait_on_log_flushed();
 
-	start_timestamp = timestamp_get();
+	do {
+		cnt = 0;
+		start_timestamp = timestamp_get();
 
-	while (start_timestamp == timestamp_get()) {
-#if (CONFIG_ARCH_POSIX)
-		k_busy_wait(100);
-#endif
-	}
+		while (start_timestamp == timestamp_get()) {
```

Comment:
I find the purpose of this uncommented logic hard to understand. Would you mind moving it out to a helper function with a clear name?

Reply:
added helper function.